src/Xamarin.Android.Build.Tasks/Tasks/GetMonoPlatformJars.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (C) 2011 Xamarin, Inc. All rights reserved.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Collections.Generic;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.Android.Tasks
+{
+	public class GetMonoPlatformJars : Task
+	{
+		[Required]
+		public string[] TargetFrameworkDirectories { get; set; }
+
+		[Required]
+		public string[] JarFiles { get; set; }
+
+		[Output]
+		public ITaskItem[] MonoPlatformJarPaths { get; set; }
+
+		[Output]
+		public ITaskItem[] MonoPlatformDexPaths { get; set; }
+
+		public override bool Execute ()
+		{
+			List<string> foundJars = new List<string> ();
+			List<string> foundDexes = new List<string> ();
+			foreach (var dir in TargetFrameworkDirectories) {
```

Comment:
This loop, along with `@(_PlatformJars)`, implies that `java_runtime.jar` needs to be in `$(TargetFrameworkDirectory)`, alongside `mono.android.jar`/`mono.android.dex`.

This feels wasteful, as we should only need *one* `java_runtime.jar`, not one per `$(TargetFrameworkVersion)`, *probably* within `$prefix/lib/xamarin.android/xbuild/Xamarin/Android`, alongside all the other MSBuild-related files.

Reply:
We have to compile our `java_runtime.jar` against the platform `.jar`. How will that work on older abi's if we just compile against the latest platform `.jar` file?

Reply:
@dellis1972: You do the opposite: you just compile against the *earliest* `android.jar` file.  Android ensures backward compatibility; code compiled for an older API level will run on newer API levels.